app/controllers/attachments_controller.rb

```diff
@@ -117,6 +118,17 @@ def attachment_params
   end
 
   def attachment_valid?
-    !params["attachment"].nil? && !params["attachment"]["file"].nil?
+    if !params["attachment"].nil? && !params["attachment"]["file"].nil?
```

Comment:
This can return `true`, `false` or `nil`, I think it would be more idiomatic to return simply `true` or `false`. You could also condense it to:
```ruby
attachment = params["attachment"]
attachment.present? && attachment["file"].present? && Attachment.valid_filetype?(attachment["file"])
```
which makes the three conditions always return a boolean.